core/src/components/loading/loading.tsx

```diff
@@ -185,6 +258,39 @@ export class Loading implements ComponentInterface, OverlayInterface {
     this.dismiss(undefined, BACKDROP);
   };
 
+  /**
+   * Determines whether or not an overlay
+   * is being used inline or via a controller/JS
+   * and returns the correct delegate.
+   * By default, subsequent calls to getDelegate
+   * will use a cached version of the delegate.
+   * This is useful for calling dismiss after
+   * present so that the correct delegate is given.
+   */
+  private getDelegate(force = false) {
```

Comment:
~ With this logic being 1:1 to modal, thoughts on abstracting this into an `overlays/utils` or similar, so implementations would call:

```ts
const { delegate } = this.getDelegate(this);
```

It would assume that `inline`, `hasController`, `workingDelegate`, `coreDelegate` and `delegate` are all members of the class.

Reply:
Updated: https://github.com/ionic-team/ionic-framework/pull/26067/files#diff-e6fe3d2c2375aaff02644f8767f36a3c49825d333f9764644871b0541a95b296R617-R675

Let me know your thoughts 👍